### PR TITLE
Fixes notification events that evaluate to multi character strings.

### LINKED ISSUE #75: Booleans transcribed into marbles cause Invalid Range Value
When testing the emission of booleans, the Marblizer turns provided booleans into `(true)` and `(false)`. However, the `start` index of the next emission is not bumped. Consider the following:

### Working reproduction
```typescript
it('breaks down', () => {
  const request$ = cold('a----b-----|');
  const resolve$ = cold('---c----d--|');
  const expected = cold('t--f-t--f--|', { t: true, f: false });

  const loading$ = merge(request$.pipe(mapTo(true)), resolve$.pipe(mapTo(false)));

  expect(loading$).toBeObservable(expected);
});
```
This errors:
```bash
    RangeError: Invalid count value
        at String.repeat (native)

      at Function../index.ts.Marblizer.marblize (../../../node_modules/jest-marbles/umd/webpack:/JestMarbles/src/marblizer.ts:14:56)
```

When stepping into the code, it turns out that `emissions` generated by `Marblizer#getNotifications` turns the marbles into the following emissions:

```typescript
[
  {
    "start": 0,
    "end": 6,
    "marbles": "(true)"
  },
  {
    "start": 3,
    "end": 10,
    "marbles": "(false)"
  },
  {
    "start": 5,
    "end": 11,
    "marbles": "(true)"
  },
  {
    "start": 8,
    "end": 15,
    "marbles": "(false)"
  },
  {
    "start": 11,
    "end": 12,
    "marbles": "|"
  }
]
```

The emission `end` is assigned to `prevEndFrame` in a for loop. Each iteration, the `TimeFrame` is repeated `emission[i].start - prevEndFrame` times. This will work for the first interation (where `prevEndFrame === 0`, but for the next iteration, `emission[i].start /* (3) */ - prevEndFrame /* (6) */` will be negative.

### PR DESCRIPTION
**Pull request by WouterSpaak**
Fixes #75 .

This should do the trick. I ran prettier and tslint on my commit which sort of mangled some spacing and imports. I can revert those chunks but then Husky won't allow me to commit.

### CODE REVIEW

**Comment by WouterSpaak** [comment]
This fix will not work actually, I still had the one test with `true`, `false` and `null` focused.

I'm thinking that whenever an object is passed to `cold(marbles, events, error?)`, the events object should just be passed through to `TestScheduler.expectObservable(obs$).toBe(marbles, events, error?)` without alteration. That's the way jasmine-marbles does it, and it works fine with jasmine. But that would require a complete rewrite of `Marblizer`.

I'll look into this a bit more.

**Comment by WouterSpaak** [comment]
> Actually it is kind of already [done](https://github.com/meltedspark/jest-marbles/blob/master/src/jest/custom-matchers.ts#L19) this way.
> I just didn't consider primitives (which in this case should be treated as objects).
> So essentially the check should be whether there is something that is not a single char among the notifications. If so, the whole thing shouldn't be marblized. And no change required in the Marblizer.

That's great news. I reverted my initial fix and implemented a `shouldMarblize` function which does just that. All tests (also the one I added that broke marblization) run just fine now.

File: src/jest/custom-matchers.ts [hunk 1]
```diff
@@ -4,33 +4,26 @@ import { SubscriptionLog } from 'rxjs/internal/testing/SubscriptionLog';
 import { TestMessage } from 'rxjs/internal/testing/TestMessage';
 import { Marblizer } from '../marblizer';
 
-function haveValueObjects(actual: TestMessage[], expected: TestMessage[]) {
-  return (
-    actual.some(m => isObject(m.notification.value) || isLiteral(m.notification.value)) ||
-    expected.some(m => isObject(m.notification.value) || isLiteral(m.notification.value))
-  );
+function shouldMarblize(...messages: TestMessage[][]) {
+  return messages.map(message => message.some(m => marblizeCheck(m))).includes(true);
```
[the remark this answers is not in the document]
Comment (WouterSpaak) [reply src/jest/custom-matchers.ts]: No, I swapped around the implementation in line 22-26 so it 'reads' a bit better.

[the remark this answers is not in the document]
Comment (meltedspark) [reply src/jest/custom-matchers.ts]: I'm probably missing something... The check appears to say the following:
"The TestMessages should be marblized if one of the Notifications in one of the TestMessages meets the requirement of marblizeCheck". Which, in turn, says:  
"The check passes if the notification value is a string and its length is 1".  

Combining these two together gives this:
"The TestMessages should be marblized if one of the Notifications in one of the TestMessages is a string and its length is 1".  

This array of TestMessages `cold('-a--b--c-', {b: true})` meets this requirement and therefore will be marblized, which will result in error.  

What am I missing?

**Comment by WouterSpaak** [comment]
Thank you for pushing changes, I did not have time to look into this more the last few days. 

Everything looks good to me!

**Pull request merged**